src/lib.rs

```diff
@@ -79,12 +81,11 @@ impl<T> ToSql<sql_types::Jsonb, Pg> for Json<T>
 where
     T: std::fmt::Debug + Serialize,
 {
-    fn to_sql<W: std::io::Write>(
-        &self,
-        out: &mut diesel::serialize::Output<W, Pg>,
-    ) -> diesel::serialize::Result {
-        let value = serde_json::to_value(self)?;
-        <serde_json::Value as ToSql<sql_types::Jsonb, Pg>>::to_sql(&value, out)
+    fn to_sql(&self, out: &mut diesel::serialize::Output<Pg>) -> diesel::serialize::Result {
+        out.write_all(&[1])?;
+        serde_json::to_writer(out, self)
+            .map(|_| IsNull::No)
+            .map_err(Into::into)
```

Comment:
Sorry to barge in, I was just looking to see if somebody had already created a diesel-2 fork 😀 Thank you for that!
```suggestion
        let value = serde_json::to_value(self)?;
        <serde_json::Value as ToSql<sql_types::Jsonb, Pg>>::to_sql(&value, &mut out.reborrow())
```

Reply:
Thanks for that suggestion!